src/backend/replication/logical/logical.c

```diff
@@ -137,7 +137,9 @@ StartupDecodingContext(List *output_plugin_options,
 	 * (re-)load output plugins, so we detect a bad (removed) output plugin
 	 * now.
 	 */
-	LoadOutputPlugin(&ctx->callbacks, NameStr(slot->data.plugin));
+
+	if (strlen(slot->data.plugin.data))
```

Comment:
The plugin is not dynamically linked any more. We should probably replace this hack with a hook.